cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs

```diff
@@ -274,43 +321,25 @@ pub async fn fetch_releases(
         let releases = releases.clone();
 
         async move {
-            trace!("[{}] Fetching release", tag);
-            let (tag, manifest, manifestref) =
-                get_manifest_and_ref(tag, repo.to_owned(), &registry_client).await?;
-
-            // Try to read the architecture from the manifest
-            let arch = match manifest.architectures() {
-                Ok(archs) => {
-                    // We don't support ManifestLists now, so we expect only 1
-                    // architecture for the given manifest
-                    ensure!(
-                        archs.len() == 1,
-                        "[{}] broke assumption of exactly one architecture per tag: {:?}",
-                        tag,
-                        archs
-                    );
-                    archs.first().map(std::string::ToString::to_string)
-                }
-                Err(e) => {
-                    error!(
-                        "could not get architecture from manifest for tag {}: {}",
-                        tag, e
+            let (arch, manifestref, mut layers_digests) =
+                get_manifest_layers(tag.to_owned(), &repo, &registry_client).await?;
+
+            // if the image is multi arch, we will have to get one image from the manifest list and
+            // use its metadata as manifest list is just a collection of manifests
```

Comment:
nit:

```rust
// if the image is multi arch, we will have to get one image from the manifest list and
// use its metadata, because manifest lists are just collections of manifests and don't
// have their own layers with metadata files.
```

or some such.  Mostly because my first read of "use its metadata as..." left me expecting more about how we would consume the metadata, but in fact you are pivoting to explain why we need to get the metadata from the referenced image, and not talking about what we do with the metadata once we've found it.